### PR TITLE
219 redo live ranking module

### PR DESCRIPTION
**Pull request by araszka**

### CODE REVIEW

File: src/Modules/LiveRankingModule/Config/ILiveRankingSettings.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    public int MaxWidgetRows { get; set; }
+    
+    [Option(DefaultValue = 63.0), Description("Specifies the Y position of the widget.")]
+    public double Y { get; set; }
+    
+    [Option(DefaultValue = 36.0), Description("Specifies the width of the widget.")]
+    public double Width { get; set; }
+    
+    [Option(DefaultValue = "right"), Description("Specifies on which side the widget is displayed.")]
+    public string Position { get; set; }
```
Comment (snixtho) [inline src/Modules/LiveRankingModule/Config/ILiveRankingSettings.cs:20]: Make this an enum instead?

File: src/Modules/LiveRankingModule/Services/LiveRankingService.cs [hunk 2]
```diff
@@ -1,259 +1,80 @@
...
-    private async Task HideManialinkAsync()
+    public Task<IEnumerable<LiveRankingPosition>> MapScoresAsync(ScoresEventArgs scores)
     {
-        await manialinkManager.HideManialinkAsync("LiveRankingModule.LiveRanking");
+        return Task.FromResult(
+            scores.Players.Take(settings.MaxWidgetRows)
+                .Where(score => score != null)
+                .OfType<PlayerScore>()
+                .Where(score => ScoreShouldBeDisplayedAsync(score).Result)
+                .Select(score => PlayerScoreToLiveRankingPositionAsync(score).Result)
```
Comment (snixtho) [inline src/Modules/LiveRankingModule/Services/LiveRankingService.cs:49]: if these methods are just utilities, then don't make them them async, thats unecceessary overhead and syntax.

File: src/Modules/LiveRankingModule/Templates/LiveRanking.mt [hunk 3]
```diff
@@ -1,102 +1,73 @@
...
-                           text="Live Ranking"
-                           valign="center2"
-                           textfont="{{ Font.Bold }}"
-                           textcolor="{{ Theme.LiveRankingModule_LiveRanking_Default_Text }}"
-                           textprefix="$i$t"
-                           textsize="2"
-                    />
+            <style
+                    class="lr-body-highlight"
+                    bgcolor="{{ Theme.UI_LiveRankingModule_Widget_RowBgHighlight }}"
```
Comment (snixtho) [inline src/Modules/LiveRankingModule/Templates/LiveRanking.mt:27]: is both the class and setting the bgcolor needed here? Isn't the class just doing that already?

**Review by snixtho — APPROVED** [review]

File: src/Modules/LiveRankingModule/Templates/LiveRanking.mt (hunk 3 above)
Comment (araszka) [reply src/Modules/LiveRankingModule/Templates/LiveRanking.mt:27]: Thats the definition of the class, I've moved it to a separate component.

File: src/Modules/LiveRankingModule/Config/ILiveRankingSettings.cs (hunk 1 above)
Comment (araszka) [reply src/Modules/LiveRankingModule/Config/ILiveRankingSettings.cs]: Created enum **WidgetPosition** in **EvoSC.Common.Util.Manialinks**

**Pull request merged**